src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticIndexingService.cs

```diff
@@ -53,193 +61,173 @@ public class ElasticIndexingService
             _indexingTaskManager = indexingTaskManager;
             _elasticConnectionOptions = elasticConnectionOptions.Value;
             _siteService = siteService;
+            _contentManager = contentManager;
+            _contentItemIndexHandlers = contentItemIndexHandlers;
+            _store = store;
             _contentDefinitionManager = contentDefinitionManager;
             _logger = logger;
         }
 
-        public async Task ProcessContentItemsAsync(string indexName = default)
+        public async Task ProcessContentItemsAsync(params string[] indexNames)
         {
```

Comment:
Too many changes in this one to understand in one try.

Reply:
@Skrypt there are multiple changes here. The important parts

 - Allow multiple parameters.
 - instead of having to pull lots of contents, we'll filter by content type. 
 - Use read only session to reduce the amount of caching done by yessql.

Reply:
To be honest I do know where to start because there is so many things that got added over the years to this. Everything has a meaning. Now, I need to re read both version and make a list of things that it needs to do and evaluate what you did. I can't do that in a short time.

Reply:
I would say view the file and browse the logic without worrying about comparing the two. Or view current file on the side and look at the logic side by side